server/game/cards/01_SOR/units/GuardianOfTheWhills.ts

```diff
@@ -0,0 +1,48 @@
+import AbilityHelper from '../../../AbilityHelper';
+import * as AbilityLimit from '../../../core/ability/AbilityLimit';
+import { Card } from '../../../core/card/Card';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { StateWatcherRegistrar } from '../../../core/stateWatcher/StateWatcherRegistrar';
+import { CardsPlayedThisPhaseWatcher } from '../../../stateWatchers/CardsPlayedThisPhaseWatcher';
+
+export default class GuardianOfTheWhills extends NonLeaderUnitCard {
+    private cardsPlayedThisPhaseWatcher: CardsPlayedThisPhaseWatcher;
+
+    protected override getImplementationId() {
+        return {
+            id: '4166047484',
+            internalName: 'guardian-of-the-whills'
+        };
+    }
+
+    protected override setupStateWatchers(registrar: StateWatcherRegistrar): void {
+        this.cardsPlayedThisPhaseWatcher = AbilityHelper.stateWatchers.cardsPlayedThisPhase(registrar, this);
+    }
+
+    private isFirstUpgradePlayedOnThisCopy(card: Card, adjusterSource: Card): boolean {
+        if (!adjusterSource.isUnit()) {
```

Comment:
Since adjusterSource will always be the host card of the ability we should be okay without this check

Reply:
This was to handle the fact taht the match method has two types of Card passed in, and inPlayId is only available to UnitCards.  This got rid of the warning that adjusterSource was indeed a unit -- is there a better way to do this?

Reply:
Added a Contract.assertTrue -- works now